packages/vats/src/virtual-purse.js

```diff
@@ -50,7 +51,9 @@ export const makeVirtualPurseKitIKit = (
   });
 
   const RetainRedeemI = M.interface('RetainRedeem', {
-    retain: M.callWhen(PaymentShape).optional(amountShape).returns(amountShape),
+    retain: M.callWhen(PaymentShape)
+      .optional(AmountShapeShape)
```

Comment:
The original here is definitely a bug. But why not change to

```suggestion
      .optional(M.pattern())
```
?
It would be more consistent with the existing ERTP guards.

Reply:
If you want to abstract this by having ERTP 
```js
export AmountShapeShape = M.pattern();
```
in order to encapsulate that representation decision, that would be fine. But in that case, we should also fix the `deposit`, `receive` and `burn` guards to use the clearer name as well, for consistency.

Reply:
I do like having the named shape there, and do the same for the methods for consistency.

But why define `AmountShapeShape` as simply `M.pattern()`? Isn't this more accurate and helpful?
```js
/** For parameters that are themselves an AmountShape pattern. */
export const AmountShapeShape = harden({
  brand: M.pattern(),
  value: M.pattern(),
});
```

Reply:
>  Isn't this more accurate and helpful?

It is more precise, in that it admits fewer patterns as valid. But the existing guards in ERTP already admit, for example, `M.any()` and an `optAmountShape` argument, which this more precise pattern would reject. So I'd say it is less accurate and helpful because it is overly precise. Were we to make this change consistently, some (hypothetical?) programs that are currently correct would start failing.

See #9410